src/dvc_studio_client/auth.py

```diff
@@ -108,7 +108,7 @@ def start_device_login(
     client_name: str,
     base_url: Optional[str] = None,
     token_name: Optional[str] = None,
-    scopes: Optional[list[str]] = None,
```

Comment:
@amritghimire what was the reason for this to be optional? (and things like `# type: ignore[arg-type,attr-defined]`) down the road?

Reply:
The reason scopes is optional is that if the scopes is not provided, Studio API provides a default scope. So, None value will be used at 
```py
    if scopes:
        body["scopes"] = scopes  # type: ignore[assignment]
```


I am not sure for the ignore comments. Looks like it was introduced in https://github.com/iterative/dvc-studio-client/pull/117/files#diff-2750549c63881942e03413261f67a7fdc17309b8eb13b634bba6c452f86ee7eb

Reply:
okay, there was an error then in this condition:

```python
if invalid_scopes := list(
            filter(lambda s: s.upper() not in AVAILABLE_SCOPES, scopes),
        )
```

(no check for `None`)

added back Optional and added a check for now